core/src/main/java/io/opencensus/tags/TagContextBuilder.java

```diff
@@ -67,6 +68,18 @@
   public abstract TagContext build();
 
   /**
+   * Enters the scope of code where the {@link TagContext} created from this builder is in the
+   * current context and returns an object that represents that scope. The scope is exited when the
+   * returned object is closed.
+   *
+   * @return an object that defines a scope where the {@code TagContext} created from this builder
+   *     is set to the current context.
+   */
+  public final Scope buildScoped() {
```

Comment:
I still have reservations about intermixing CurrentTagContext in with TagContext{Builder}, let's talk about it at our next sync (tomorrow).